### PR TITLE
style-minify

### LINKED ISSUE #507: Optimize css in production
Probably using https://github.com/NMFR/optimize-css-assets-webpack-plugin, or https://github.com/sandark7/csso-loader

### PR DESCRIPTION
**Pull request by timkelty**
Resolves #507

### CODE REVIEW

File: packages/style-minify/README.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# Neutrino Style Minify Middleware
```
Comment (edmorley) [inline packages/style-minify/README.md]: Hi! Thank you for the PR :-)
Could you copy this README to docs/ too? (I'm hoping we can do away with this duplication soon)

File: packages/style-minify/README.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+// Use with default options
+module.exports = {
+  use: ['@neutrinojs/style-minify']
+};
+
+// Usage showing overriding options
+module.exports = {
+  use: [
+    ['@neutrinojs/style-minify', {
+    }]
```
Comment (eliperelman) [inline packages/style-minify/README.md]: Missing overriding options here.

File: packages/style-minify/README.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  ]
+};
+```
+
+The `plugin` property is passed to [optimize-css-assets-webpack-plugin
```
Comment (eliperelman) [inline packages/style-minify/README.md]: Make this one line or move the whole url anchor to the second line.

File: packages/style-minify/README.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+`@neutrinojs/style-minify` creates some conventions to make overriding the configuration easier once you are ready to
+make changes.
+
+### Plugins
+
+The following is a list of plugins and their identifiers which can be overridden:
+
+| Name | Description | Environments and Commands |
+| --- | --- | --- |
+| `style-minify` | Minifies styles using `OptimizeCssAssetsPlugin`. | all |
```
Comment (eliperelman) [inline packages/style-minify/README.md]: I just realized that we need to update the web preset docs to add these plugins as well.

File: packages/style-minify/index.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+const merge = require('deepmerge');
+const OptimizeCssPlugin = require('optimize-css-assets-webpack-plugin');
+
+module.exports = ({ config }, opts = {}) => {
+  const options = merge({
+    pluginUseId: 'style-minify',
```
Comment (eliperelman) [inline packages/style-minify/index.js]: This should just be `pluginId`.

File: packages/style-minify/README.md (hunk 1 above)
Comment (timkelty) [reply packages/style-minify/README.md]: of course, forgot!

File: packages/style-minify/README.md (hunk 4 above)
Comment (timkelty) [reply packages/style-minify/README.md]: Should already be good to go…web doesn't use them separately, it uses `@neutrinojs/minify` which has them all.

**Comment by timkelty** [comment]
Ok, I think everyone's review comments are addressed.

File: packages/style-minify/README.md (hunk 4 above)
Comment (eliperelman) [reply packages/style-minify/README.md]: You'll notice the minify plugin is out of date:

https://github.com/mozilla-neutrino/neutrino-dev/tree/master/packages/web#plugins

For these core presets, we have listed out all the transitive plugins and loaders.

File: packages/style-minify/index.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+const OptimizeCssPlugin = require('optimize-css-assets-webpack-plugin');
+
+module.exports = ({ config }, opts = {}) => {
+  const options = merge({
+    pluginId: 'style-minify',
+    plugin: {}
+  }, opts);
+
+  config
+    .plugin(options.pluginUseId)
```
Comment (eliperelman) [inline packages/style-minify/index.js]: Need to update this to `pluginId` as well.

File: packages/style-minify/README.md (hunk 4 above)
Comment (timkelty) [reply packages/style-minify/README.md]: Ah i see

File: packages/style-minify/index.js (hunk 6 above)
Comment (timkelty) [reply packages/style-minify/index.js]: (╯°□°）╯︵ ┻━┻

Comment (eliperelman) [reply packages/style-minify/index.js]: lololol

File: packages/style-minify/README.md (hunk 4 above)
Comment (eliperelman) [reply packages/style-minify/README.md]: Are you doing that in this PR, or a should we file a follow up?

Comment (timkelty) [reply packages/style-minify/README.md]: I'll do it here.

File: docs/packages/web/README.md [hunk 7]
```diff
@@ -356,7 +356,9 @@ _Note: Some plugins are only available in certain environments. To override them
 | `hot` | Enables Hot Module Replacement. From `@neutrinojs/hot`. | `start` command |
 | `copy` | Copies files during build, defaults from `src/static` to `build/static`. From `@neutrinojs/copy` | `build` command |
 | `clean` | Removes the `build` directory prior to building. From `@neutrinojs/clean`. | `build` command |
-| `minify` | Minifies source code using `BabelMinifyWebpackPlugin`. From `@neutrinojs/minify`. | `NODE_ENV production` |
+| `babel-minify` | Minifies source code using `BabelMinifyWebpackPlugin`. From `@neutrinojs/minify`. | `NODE_ENV production` |
+| `imagemin` | Optimize any images added by other webpack plugins (e.g. `copy-webpack-plugin`). | `NODE_ENV production` |
```
Comment (eliperelman) [inline docs/packages/web/README.md]: Missing the from here. "From `@neutrinojs/minify`."

File: docs/packages/web/README.md [hunk 8]
```diff
@@ -356,7 +356,9 @@ _Note: Some plugins are only available in certain environments. To override them
...
+| `optimize-css` | Minifies css using `OptimizeCssAssetsPlugin`. | `NODE_ENV production` |
```
Comment (eliperelman) [inline docs/packages/web/README.md]: Missing the from here. "From `@neutrinojs/minify`."

File: packages/web/README.md [hunk 9]
```diff
@@ -356,7 +356,9 @@ _Note: Some plugins are only available in certain environments. To override them
 | `hot` | Enables Hot Module Replacement. From `@neutrinojs/hot`. | `start` command |
 | `copy` | Copies files during build, defaults from `src/static` to `build/static`. From `@neutrinojs/copy` | `build` command |
 | `clean` | Removes the `build` directory prior to building. From `@neutrinojs/clean`. | `build` command |
-| `minify` | Minifies source code using `BabelMinifyWebpackPlugin`. From `@neutrinojs/minify`. | `NODE_ENV production` |
+| `babel-minify` | Minifies source code using `BabelMinifyWebpackPlugin`. From `@neutrinojs/minify`. | `NODE_ENV production` |
+| `imagemin` | Optimize any images added by other webpack plugins (e.g. `copy-webpack-plugin`). | `NODE_ENV production` |
```
Comment (eliperelman) [inline packages/web/README.md]: Missing the from here. "From `@neutrinojs/minify`."

File: packages/web/README.md [hunk 10]
```diff
@@ -356,7 +356,9 @@ _Note: Some plugins are only available in certain environments. To override them
...
+| `optimize-css` | Minifies css using `OptimizeCssAssetsPlugin`. | `NODE_ENV production` |
```
Comment (eliperelman) [inline packages/web/README.md]: Missing the from here. "From `@neutrinojs/minify`."

**Comment by timkelty** [comment]
@eliperelman Think docs should be good now!

**Comment by eliperelman** [comment]
Amazing!

**Pull request merged**